src/compositemark.ts

```diff
@@ -72,3 +79,175 @@ add(ERRORBAR, (spec: GenericUnitSpec<ERRORBAR, Encoding>): LayerSpec => {
     ]
   };
 });
+
+add(BOX, (spec: GenericUnitSpec<BOX, Encoding>): LayerSpec => {
+  const {mark: _m, encoding: encoding, ...outerSpec} = spec;
+  const {x: _x, y: _y, ...nonPositionEncoding} = encoding;
+
+  let fullEncoding: any = {};
+
+  let isVertical: boolean;
+
+  if (isDiscrete(encoding.x) && isContinuous(encoding.y)) {
```

Comment:
Tip: you can use the [computed properties syntax](https://github.com/Microsoft/TypeScript/wiki/What's-new-in-TypeScript#computed-properties) do something like 

```ts
return {
  ...outerSpec,
  layer: [{
    mark: 'rule',
    encoding: {
       [discreteAxisChannel]: <...some code...>,
       [continuousAxisChannel]: <...some code..>,
    },
    <...more code...>
  ]
}

```